crates/kubelet/src/node.rs

```diff
@@ -38,11 +38,28 @@ pub async fn create_node(client: &kube::Client, config: &Config, arch: &str) {
                 e
             );
             match node_client.get(&config.node_name).await {
-                Ok(node) => {
+                Ok(_) => {
+                    // HACK WARNING: So it turns out we need to have the proper
+                    // permissions in order to update the node status, so this
+                    // is a hacky workaround for now where we delete and
+                    // recreate the node. This is being tracked in https://github.com/deislabs/krustlet/issues/150
                     info!("node found, updating current node definition");
-                    update_node(
-                        client,
-                        &node.metadata.unwrap_or_default().name.unwrap_or_default(),
+                    node_client
+                        .delete(&config.node_name, &DeleteParams::default())
+                        .await
+                        .expect("Unable to recreate node...aborting");
```

Comment:
Should this function return a `Result` instead of panicing on these? This means we'll crash the program on network failure.

Reply:
This should panic here because it wasn't able to re-establish control over the node, leaving things in a completely unknown state

Reply:
Hmm.. this is the first write action we perform in the error recovery arm. Shouldn't the caller be able to recover?

Reply:
The proper thing here would be a retry. I'll add an issue to 0.2 for it

Reply:
Created #155